.github/workflows/ubuntu.yml

```diff
@@ -109,125 +94,102 @@ jobs:
                       --with-cplex-incdir=$DOWNWARD_CPLEX_ROOT/include/ilcplex \
                       --with-cplex-lib="-lcplex -lm -ldl" # -ldl is only needed for CPLEX >= 12.8
-          make
+          make -j2
           make install
           cd ../
           rm -r Osi-0.107.9.tgz Osi-0.107.9
 
-      - name: compile
-        env:
-          DOWNWARD_CPLEX_ROOT: /home/runner/work/downward/lib/ibm/ILOG/CPLEX_Studio129/cplex
-          DOWNWARD_SOPLEX_ROOT: /home/runner/work/downward/lib/soplex-3.1.1
-          DOWNWARD_COIN_ROOT: /home/runner/work/downward/lib/coin
+      - name: Compile planner
         run: |
-          export CXXFLAGS="-Werror"  # Treat compilation warnings as errors.
+          export CXXFLAGS="-Werror" # Treat compilation warnings as errors.
           ./build.py --debug
           ./build.py
 
-      - name: setup-uncrustify
-        run: |
-          # Set up uncrustify.
-          wget https://github.com/uncrustify/uncrustify/archive/uncrustify-0.67.tar.gz
-          tar xzf uncrustify-0.67.tar.gz
-          cd uncrustify-uncrustify-0.67
-          mkdir build
-          cd build
-          cmake ../
-          make -j8
-          mv uncrustify ../../
-          cd ../../
-          rm -rf uncrustify-0.67.tar.gz uncrustify-uncrustify-0.67
-
-      - name: setup-val
-        run: |
-          # Set up VAL.
-          sudo apt-get -y install flex bison
-          git clone https://github.com/KCL-Planning/VAL.git
-          cd VAL
-          git checkout a5565396007eee73ac36527fbf904142b3077c74
-          make clean  # Remove old build artifacts and binaries.
-          sed -i 's/-Werror //g' Makefile  # Ignore warnings.
-          make -j8
-          mv validate ../
-          cd ../
-          ls -la
-          rm -rf VAL
-
-      - name: archive-files
+      - name: Archive required files
         # We only run tests on the version compiled with gcc, so we
         # only need to archive that one.
-        if: ${{ matrix.compiler-version }} == gcc
+        if: ${{ matrix.compiler-version.cc == 'gcc' }}
+        # We determined the dynamically linked libraries using ldd.
         run: |
           cd ../
-          tar cfz ${{ matrix.compiler-version }}.tar.gz downward lib
+          tar cfz archive.tar.gz downward/builds/debug/bin/ downward/builds/release/bin/ lib/coin/lib/libOsiCpx.so.1 lib/ibm/ILOG/CPLEX_Studio129/cplex/bin/x86-64_linux/libcplex1290.so lib/coin/lib/libOsiSpx.so.1 lib/coin/lib/libOsi.so.1 lib/coin/lib/libCoinUtils.so.3
 
-      - name: upload-files
-        if: ${{ matrix.compiler-version }} == gcc
-        uses: actions/upload-artifact@v1
+      - name: Upload archive
+        if: ${{ matrix.compiler-version.cc == 'gcc' }}
+        uses: actions/upload-artifact@master
         with:
           name: compiled-planner-${{ matrix.ubuntu-version }}
-          path: /home/runner/work/downward/${{ matrix.compiler-version }}.tar.gz
+          path: /home/runner/work/downward/archive.tar.gz
```

Comment:
Let's add a retention time limit of 1 day and/or delete the artifact at the end of the workflow.